### DISCUSSION TITLE
Support for Same application on multiple environments

### DISCUSSION THREAD

**Discussion by vpandi** [argoproj/argo-cd#9832] [General]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**
We want to go with the centralised argocd setup which ultimately create 1500+ argocd application CRDs in multi region data centres.
Do we have any constraint in creating numerous applications in a single EKS cluster ?

**To Reproduce**

1500 Applications in a single argo cluster 

When we worked on other project we had an issue with our control plane with heavy CRDs, Want to understand if any one used argo CD for such a huge number of applications in a single argo cluster and any pointers on the performance side also helps.
Please advice. 

**Version**

```$ argocd version --short
argocd: v2.0.4+0842d44
argocd-server: v2.3.1+b65c169
```

**Logs**

```
No logs
```

**Comment by jessesuen** [comment]
1500 should not be a problem (we have users who 3x that) but UI is known to be sluggish because filtering/paging is done client side ATM and could benefit from server side pagination.

**Comment by vpandi** [comment]
Thanks @jessesuen . Let me ask a follow up question. We have 150+ micro services which uses helm chart. We do have custom values based on the environment such dev,qa,stage and prod. We assume we need to create a different argocd application for every app in every environment. Letus demo is our microservice name

So we need to create argo application such as  demodev,demoqa  and demostage and so on.
Every time we need to deploy demo service in dev data center we need to sync the demodev argocd application. Is our understanding is correct?

With the above understanding only we assumed that it is going to be 1500+ argocd application in single cluster.
